rapids-4-spark-tools/src/test/scala/com/nvidia/spark/rapids/tool/profiling/ApplicationInfoSuite.scala

```diff
@@ -56,9 +56,62 @@ class ApplicationInfoSuite extends FunSuite with Logging {
       assert(apps.head.stageSubmitted(2).stageId.equals(2))
       assert(apps.head.taskEnd(apps.head.index).successful.equals(true))
       assert(apps.head.taskEnd(apps.head.index).endReason.equals("Success"))
+      assert(apps.head.executors.head.totalCores.equals(8))
+      println(apps.head.resourceProfiles.head.exec_mem)
+      assert(apps.head.resourceProfiles.head.exec_mem.equals(1024L))
     } finally {
       fileWriter.close()
       tempFile.deleteOnExit()
     }
   }
+
+  test("test filename match") {
+    val matchFileName = "udf"
+    val appArgs = new ProfileArgs(Array(
+      "--match-event-logs",
+      matchFileName,
+      "src/test/resources/udf_func_eventlog",
+      "src/test/resources/udf_dataset_eventlog",
+      "src/test/resources/dataset_eventlog"
+    ))
+
+    val result = ProfileMain.processAllPaths(appArgs.filterCriteria,
+      appArgs.matchEventLogs, appArgs.eventlog())
+    assert(result.length == 2)
+  }
+
+  test("test filter file newest") {
+    val filterNew = "2-newest"
+    val appArgs = new ProfileArgs(Array(
+      "--filter-criteria",
+      filterNew,
+      "src/test/resources/udf_func_eventlog",
+      "src/test/resources/udf_dataset_eventlog",
+      "src/test/resources/dataset_eventlog",
+      "src/test/resources/eventlog_minimal_events"
+    ))
+
+    val result = ProfileMain.processAllPaths(appArgs.filterCriteria,
+      appArgs.matchEventLogs, appArgs.eventlog())
+    assert(result.length == 2)
+  }
+
+  test("test filter file oldest and file name match") {
+    val filterOld = "3-oldest"
+    val matchFileName = "event"
+    val appArgs = new ProfileArgs(Array(
+      "--filter-criteria",
+      filterOld,
+      "--match-event-logs",
+      matchFileName,
+      "src/test/resources/udf_func_eventlog",
+      "src/test/resources/udf_dataset_eventlog",
+      "src/test/resources/dataset_eventlog",
+      "src/test/resources/eventlog_minimal_events"
+    ))
+
+    val result = ProfileMain.processAllPaths(appArgs.filterCriteria,
+      appArgs.matchEventLogs, appArgs.eventlog())
+    assert(result.length == 3)
```

Comment:
similar validate that we got the correct files back.

Reply:
Updated the test.